### ISSUE TITLE
Text field character counter alert and limit

### ISSUE THREAD

**Issue by dnvargas25** [wwnorton/design-system#11]
Based on the Storybook example, the [character counter text field ](https://wwnorton.github.io/design-system/storybook/?path=/story/text-field--with-max-length) might be challenging for users, particularly for screen reader users, because the character count and limit isn't clear to all users. 

- When a user types in this field, the maximum number of characters is not announced to screen readers. Without announcing the character count, these users will need to discover the count on their own (which isn't available until the user types two characters in the text field). On the other hand, alerting screen reader users to the new character count ("#/# characters remaining") after every keystroke will likely be more annoying than helpful. Maybe there is a solution to provide updates to screen reader users without bombarding them with alerts. 
- When a user reaches the maximum number of characters in this text field, the user can technically continue to type, but the keystrokes are not reflected in the text field. This could be problematic for users who want to paste a larger block of text than is allowed or users who aren't looking (or can't look) at the screen. 

These ideas are reflected in the Docs: [under the heading "Usage"](https://wwnorton.github.io/design-system/docs/components/text-field#usage), they're the last two bullets in the "Character Counter" box, but these ideas aren't reflected in the Storybook example.

**Comment by sh0ji** [comment]
For reference, the character counter's use of the HTML [`maxLength`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength) results in us not following our own guidance ("Allow users to enter more than the character limit, and then display an error. This lets a user type or copy and paste their full text entry, then edit it down."). That should definitely be fixed.

Fixing that would also resolve the screen reader experience, as our error feedback should trigger after the user gets to `0/max` characters remaining.
